src/codeu/chat/client/maingui/ConversationPanel.java

```diff
@@ -0,0 +1,136 @@
+    titleLabel.setAlignmentX(Component.LEFT_ALIGNMENT);
+    titlePanel.add(titleLabel);
+
+    // Conversation list
+    final JPanel listShowPanel = new JPanel(new BorderLayout());
+
+    final DefaultListModel<String> listModel = new DefaultListModel<>();
+    final JList<String> objectList = new JList<>(listModel);
+    objectList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+    objectList.setSelectedIndex(-1);
+
+    final JScrollPane listScrollPane = new JScrollPane(objectList);
+    listShowPanel.add(listScrollPane, BorderLayout.CENTER);
+
+    // Button bar
+    final JPanel buttonPanel = new JPanel();
+
+    final JButton updateButton = new JButton("Update");
+    final JButton addButton = new JButton("Add");
+
+    updateButton.setAlignmentX(Component.LEFT_ALIGNMENT);
+    buttonPanel.add(updateButton);
+    buttonPanel.add(addButton);
+
+    this.add(titlePanel, BorderLayout.NORTH);
+    this.add(listShowPanel, BorderLayout.CENTER);
+    this.add(buttonPanel, BorderLayout.SOUTH);
+
+    // User clicks Conversations Update button.
+    updateButton.addActionListener(new ActionListener() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        ConversationPanel.this.getAllConversations(listModel);
+      }
+    });
+
+    // User clicks Conversations Add button.
+    addButton.addActionListener(new ActionListener() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        if (clientContext.user.hasCurrent()) {
+          final String s = (String) JOptionPane.showInputDialog(ConversationPanel.this, "Enter title:", "Add Conversation", JOptionPane.PLAIN_MESSAGE,null, null, "");
+          if (s != null && s.length() > 0) {
+            clientContext.conversation.startConversation(s, clientContext.user.getCurrent().id);
+            ConversationPanel.this.getAllConversations(listModel);
+            
+            messagePanel.update(clientContext.conversation.getCurrent());
+          }
+        } else {
+          JOptionPane.showMessageDialog(ConversationPanel.this, "You are not signed in.");
+        }
+      }
+    });
+
+    // User clicks on Conversation - Set Conversation to current and fill in Messages panel.
+    objectList.addListSelectionListener(new ListSelectionListener() {
+      @Override
+      public void valueChanged(ListSelectionEvent e) {
+        if (objectList.getSelectedIndex() != -1) {
+          final int index = objectList.getSelectedIndex();
+          final String data = objectList.getSelectedValue();
+          final ConversationSummary cs = ConversationPanel.this.lookupByTitle(data, index);
+
+          clientContext.conversation.setCurrent(cs);
+          messagePanel.update(cs);
+        }
+      }
+    });
+
+    getAllConversations(listModel);
+  }
+
+  // Populate ListModel - updates display objects.
+  private void getAllConversations(DefaultListModel<String> convDisplayList) {
+
+    clientContext.conversation.updateAllConversations(false);
```

Comment:
it's often helpful to put an inline comment with the variable name that the true/false is for, e.g.,

clientContext.conversation.updateAllConversations(false /* currentChanged */);

(Though, I still don't know what it's for given that variable name...)